File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.py
class Solution:
    def nextGreatestLetter(self, A: List[str], t: str) -> str:
        start,end = 0, len(A)
        while start < end:
            mid = start + (end - start) // 2 
            if(A[mid] > t):
                end = mid
            else:
                start = mid + 1
        return A[start % len(A)]

File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& l, char t) {
       int start(0), end(l.size());
        while (start < end)
        {
            int mid = start + (end - start) / 2;
            if(l[mid] > t) end = mid;
            else start = mid + 1;
        }
        return l[start % l.size()] ;
    }
};